### DISCUSSION TITLE
FileNotFound error on Windows

### DISCUSSION THREAD

**Discussion by sleepingcat4** [openai/whisper#61] [Q&A] [answered]
In Initial testing, I wanted to translate/transcribe some audio files and copy pasted the code that was written in the documentation, unfortunately, it shows that it can't find the file. Even the though the audio files and the python script are in the same directory. 

is there any catch, I'm missing out?

**Comment by FurkanGozukara** [comment]
move to the same folder of your sound in cmd. then it works

**Reply by sleepingcat4** [reply]
I'm on VS code and upon running the python script it automatically runs in the folder I'm in. 

And the audio and python script are in the same folder (there's no separate subfolder for the audio file)

**Reply by FurkanGozukara** [reply]
I have run through CMD it worked but only after when I entered same folder.

**Reply by sleepingcat4** [reply]
I think you didn't understand my question, I am using python to run the model not CMD

**Reply by junlanlu** [reply]
I had the same issue, it was the result of not having ffmpeg properly installed.

**Reply by sleepingcat4** [reply]
> I had the same issue, it was the result of not having ffmpeg properly installed.

I believe so, but I have installed FFMPEG before on my computer as part of other projects so kinda confused, will reinstall with latest version using Chocolatey

**Comment by fcakyon** [comment]
@sleepingcat4 @junlanlu ffmpeg cli setup can be troublesome, I have created a whisper package that does not require manual ffmpeg installation. Feel free to try it: https://github.com/fcakyon/pywhisper

**Reply by sleepingcat4** [reply]
Actually I had previously installed FFMEG on my computer as part of a deep learning project. That's why, I was taken aback why the error was showing itself but thanks for the repo!

**Reply by Dirk-Vantas** [reply]
thank you for preventing my last 2 braincells from committing toaster bath with your repo

**Reply by Tkag0001** [reply]
> @sleepingcat4 @junlanlu ffmpeg cli setup can be troublesome, I have created a whisper package that does not require manual ffmpeg installation. Feel free to try it: https://github.com/fcakyon/pywhisper

It's very helpful. In my case I downloaded ffmpeg and tested it on common cmd. It is available but when I check cmd in a project with individual environment it is not available. I tried several methods but still couldn't solve it. Thank you for pywhisper.

**Comment by hbarnard** [comment]
Thanks, having the same problem but on replicate.ai, so a local library can't solve it. I've put some debugging into where it happens:

```
>>> output = model.predict(audio="OSR_us_000_0010_8k.wav")
 Kwargs: {}
 Kwargs: {'json': {'version': '770db50964b436879e870139c9c1504d6326774d8acc92e6815c19b68367ec51', 'input': {'audio': 'OSR_us_000_0010_8k.wav'}}}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.7/dist-packages/replicate/model.py", line 20, in predict
    return latest_version.predict(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/replicate/version.py", line 18, in predict
    prediction = self._client.predictions.create(version=self, input=kwargs)
  File "/usr/local/lib/python3.7/dist-packages/replicate/prediction.py", line 72, in create
    json=body,
  File "/usr/local/lib/python3.7/dist-packages/replicate/client.py", line 55, in _request
    raise ReplicateError(resp.json()["detail"])
replicate.exceptions.ReplicateError: File does not exist: OSR_us_000_0010_8k.wav
```

**Comment by mm225022** [comment]
I am also having the same issue.  It works fine on Google Colab, but trying to run in Python directly on my machine, either in PyCharm or VSCode results in the FileNotFound issue.
Colab is nice when it's test data, but it won't work for real data, so I'd like to get resolved how I can get this to work on my machine especially as it sounds like others are having the same issue.
Thanks

**Reply by hbarnard** [reply]
I don't use colab or pycharm, but here's a working snippet from my code. Hope it helps a bit, anyway.

```
 # or use external speech to text model on replicate.com
    
        audio_file = Path(file_path)  # this was 'my' mistake!
        api = replicate.Client(api_token=my_env['REPLICATE_API_TOKEN'])
        model = api.replicate.models.get("openai/whisper")
        result = model.predict(audio=audio_file)
        text = result['transcription']        
```

**Comment by aster-fung** [comment]
I used absolute file path as a quick fix. it works. 
remember to escape the \ with \\

**Comment by bharatchandra-dev** [comment, marked as answer]
I was able to resolve the issue. It turns out that the problem was due to an improper installation of ffmpeg. Fortunately, I came across a helpful resource at [geeksforgeeks](https://www.geeksforgeeks.org/how-to-install-ffmpeg-on-windows/) that provided detailed instructions on how to install ffmpeg correctly. Following the steps outlined in the guide, I was able to install ffmpeg successfully, and it is now working perfectly on my Windows 11 system. In visual studio code also it is working now.

**Reply by andrepiper** [reply]
This is the fix. Thanks @bharatchandra-dev

**Reply by selwynluo** [reply]
This is the very right solution, thank you!

**Comment by adriy-be** [comment]
To solve the problem on my side on Window :
find the path of ffmpeg
```
Get-Command ffmpeg
CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Application     ffmpeg.exe                                         0.0.0.0    C:\Users\CHANGEME\AppData\Local\Microsoft\WinGet\Packages\Gyan.FFmpeg_Microsof...
```

And add to your script
```
os.environ['PATH']
```

For me is :
```
os.environ['PATH'] += os.pathsep + r'C:\Users\CHANGEME\AppData\Local\Microsoft\WinGet\Packages\Gyan.FFmpeg_Microsoft.Winget.Source_8wekyb3d8bbwe\ffmpeg-7.0.2-full_build\bin'
```

Now that working.
